pkg/utils/utils.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/containernetworking/cni/pkg/types"
+)
+
+// ValidInputString is the regexp used to validate valid characters in
+// containerID and networkName
+const CNIvalidNameChars = `[a-zA-Z0-9][a-zA-Z0-9_.-]`
```

Comment:
Please escape (\_) the trailing hyphen.  While _ and . are not special characters in character classes, hyphens are - although I'm unsure whether that applies at the end.

Reply:
Is the escaping needed inside ``  ?

Reply:
It's not clear: the hyphen certainly sometimes has meaning inside the character class (because it's used for the a-z ranges).

Reply:
Ack.